src/domain/challenge/challenge/challenge.service.ts

```diff
@@ -472,37 +361,34 @@ export class ChallengeService {
     return [challenge, organisation];
   }
 
-  async createApplication(
-    id: number,
-    applicationData: ApplicationInput
-  ): Promise<Application> {
-    const challenge = (await this.getChallengeOrFail(id, {
-      relations: ['applications'],
-    })) as Challenge;
-
-    const existingApplication = challenge.applications?.find(
-      x => x.user.id === applicationData.userId
+  async removeChallengeLead(
+    challengeID: number,
+    organisationID: number
+  ): Promise<boolean> {
+    let challenge;
+    // eslint-disable-next-line prefer-const
+    [challenge, {}] = await this.getChallengeAndOrganisation(
```

Comment:
This is an odd approach, have a method that returns two objects just to take one of them. This approach is more for external methods where you don't need one / more than one of the return objects, not for our own services.

Reply:
Updated code